app/src/main/java/com/portfolio/tasky/BaseActivity.kt

```diff
@@ -0,0 +1,50 @@
+package com.portfolio.tasky
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+
+open class BaseActivity : AppCompatActivity() {
+    private var mActivity: BaseActivity? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        mActivity = this
```

Comment:
Why would you have an instance of yourself?

I know it's for later. But im curious

Reply:
Since the BaseActivity and BaseFragment will serve as an extension to all other activities/fragments, it will be easy for me to access activity's context which is in foreground easily. (As per my understandings)

Reply:
I'd recommend not to use any base classes (at least if they're called `BaseX`), since that really smells like a violation of the single responsibility principle. If there's one clear responsibility for the class, you could give it a clear name which `BaseActivity` or `BaseFragment` isn't. 
Having base classes couples all classes to that base class, so any change in the base class will have an impact on all other classes which you can usually find better solutions for. What's reasonable is to have a root fragment for example for the ViewBinding setup, but then this root fragment has a clear responsibility and could be called `BindingFragment` or similar